mint/git-clone/mint-leaf.yml

```diff
@@ -109,10 +109,12 @@ tasks:
 
       unresolved_ref=""
       if [[ -n "${META_REF}" ]]; then
-        refs_matching_provided_ref=$(git for-each-ref --format="%(refname)" "refs/heads/${META_REF}" "refs/tags/${META_REF}" "${META_REF}" | grep -v refs/remotes)
+        refs_matching_provided_ref=$(git ls-remote --heads --tags origin | grep "refs/heads/${META_REF}\|refs/tags/${META_REF}\|${META_REF}" | awk '{ print $2; }')
         unresolved_ref=$(echo "$refs_matching_provided_ref" | head -n 1 | tr -d '\n')
 
         # also, ensure the meta-ref contains the resolved commit sha
+        # first fetch the ref so it will appear in git for-each-ref under refs/heads or refs/tags
+        git fetch origin "${unresolved_ref}:${unresolved_ref}"
         result=$(git for-each-ref "${unresolved_ref}" --format="%(refname)" --contains "${commit_sha}")
```

Comment:
We still use `for-each-ref` here because `ls-remote` doesn't have the commit filtering capabilities, but in this case we know the exact ref we want so we can fetch it locally first to avoid the issue this PR solves